src/basic_recipes/barplot.jl

```diff
@@ -110,7 +110,11 @@ function stack_grouped_from_to(i_stack, y, grp)
     from = Array{Float64}(undef, length(y))
     to   = Array{Float64}(undef, length(y))
 
-    groupby = StructArray((; grp..., is_pos = y .> 0))
```

Comment:
As far as I can tell just changing this to `.>=` works too.

Reply:
I am afraid not. Given two stacked bars of heights [-6, 0], the second bar would stacked on the first one as expected.

Reply:
Oh. You do remind me that we should determine the `is_pos` only based other bars inside of the same group, instead of globally. I will append a commit later. Thanks.

Reply:
Sorry for the delayed update. I have been too busy recently :(. A new commit has been added. You can test it using:
```
import Makie, CairoMakie

fig = Makie.Figure()

ax = Makie.Axis(fig[1, 1])
xs = [1, 1, 1, 1]
ys = [0, 1, -1, -1]
stack = [1, 2, 1, 2]
dodge = [1, 1, 2, 2]
color = Makie.cgrad(:Set2_8)[1:4]
Makie.barplot!(ax, xs, ys; stack, dodge, color, bar_labels=string.(ys))

display(fig)
```